app/src/main/kotlin/com/softteco/template/ui/components/PasswordField.kt

```diff
@@ -0,0 +1,130 @@
+package com.softteco.template.ui.components
+
+import androidx.compose.animation.animateColorAsState
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Visibility
+import androidx.compose.material.icons.filled.VisibilityOff
+import androidx.compose.material.icons.rounded.Check
+import androidx.compose.material.icons.rounded.Close
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import com.softteco.template.R
+import com.softteco.template.ui.feature.PasswordFieldState
+import com.softteco.template.ui.theme.Dimens
+
+@Composable
+fun PasswordField(
+    passwordValue: String,
+    onPasswordChanged: (String) -> Unit,
+    fieldStatePassword: PasswordFieldState,
+    isPasswordHasMinimum: Boolean?,
```

Comment:
It seems false could be used as the default value, then the nullable option could be removed.

Reply:
it was created to know, if  we should show validation fields or not:
```
    if (isPasswordHasMinimum != null && isPasswordHasUpperCase != null) {
        Spacer(modifier = Modifier.height(Dimens.PaddingDefault))
        Column(verticalArrangement = Arrangement.spacedBy(Dimens.PaddingDefault)) {
            ConditionRow(
                condition = stringResource(R.string.registration_password_condition1),
                check = isPasswordHasMinimum
            )
            ConditionRow(
                condition = stringResource(R.string.registration_password_condition2),
                check = isPasswordHasUpperCase
            )
        }
    }
```